Fix imu_integrator_input timestamp

qinjunj opened the pull request.

qinjunj committed:
correct imu_integrator_input timestamp

qinjunj committed:
correct imu_integrator_input timestamp

qinjunj committed:
Revert "correct imu_integrator_input timestamp"

This reverts commit b4eff07068ab8dbde13cf3b51352305fc3e763ec.

qinjunj committed:
Revert "correct imu_integrator_input timestamp"

This reverts commit b4eff07068ab8dbde13cf3b51352305fc3e763ec.

qinjunj committed:
correct imu_integrator_input timestamp

qinjunj committed:
correct imu_integrator_input timestamp

jianxiapyh commented on ov_msckf/src/slam2.cpp:
```diff
@@ -272,7 +272,7 @@ class slam2 : public plugin {
 
 			_m_imu_integrator_input.put(_m_imu_integrator_input.allocate(
-				datum->time,
+				imu_cam_buffer->time,
```
my guess is that this is wrong because imu_cam_buffer is only assigned to datum at the end of this function, so this is essentially using the next buffer's timestamp instead?

qinjunj replied:
Becasue the imu_cam_buffer is only assigned at the end of the function, so in each iteration we are using the time of the last daum. This is consistent with open_vins internally buffering the image for one iteration.

qinjunj committed:
Issue 50 separate imu cam (#15)

* Move cvtfmt into open_vins

* MTP fix is done?

* Add back prints

* Gate prints

* New Switchboard API (mem leak)

* New allocate

* Replaying merge with master

Co-authored by Sam Grayson <sam@samgrayson.me>

* Use time_point

* Fix broken merge

* Fix broken merge

* wip

* wip

* use relative clock

* separate imu and cam type

* MTP fix is done?

* Add back prints

* Gate prints

* New Switchboard API (mem leak)

* Replaying merge with master

Co-authored by Sam Grayson <sam@samgrayson.me>

* Use time_point

* wip

* use relative clock

* separate imu and cam type

* not first dummy pose

* rebase with master

* formatting

* no cast needed since the imu data is kept doubles

* remove stale comments

Co-authored-by: Samuel Grayson <sam@samgrayson.me>
Co-authored-by: Qinjun Jiang <qinjunj2@illinoise.edu>
Co-authored-by: Giordano Salvador <73959795+e3m3@users.noreply.github.com>

qinjunj committed:
Merge branch 'fix-imu-integrator-input-timestamp' of https://github.com/ILLIXR/open_vins into fix-imu-integrator-input-timestamp

qinjunj commented:
Will rebase on the new master.
